annotation-utils/src/main/java/edu/isi/vista/annotationutils/ExtractAnnotationStats.kt

```diff
@@ -0,0 +1,223 @@
+                            text-align: left;
+                            padding: 8px;
+                        }"""
+                    }
+                }
+                body {
+                    h2 { +"Annotation Statistics - $thisDate"}
+                    table {
+                        tr {
+                            th {+"Total annotations"}
+                            th {+"New annotations"}
+                        }
+                        tr {
+                            td {+"$totalAnnotations"}
+                            if (previousAnnotations == 0) {
+                                td{+"n/a"}
+                            }
+                            else {
+                                td { +"${totalAnnotations - previousAnnotations}" }
+                            }
+                        }
+                    }
+                    table {
+                        tr {
+                            th {+"User"}
+                            th {+"Annotations"}
+                            th {+"New annotations"}
+                        }
+                        for (user in sortedUsers.keys) {
+                            tr {
+                                td {+"$user"}
+                                td {+"${annotationsPerUser[user]}"}
+                                if (previousAnnotationsPerUser[user] == null) {
+                                    td {+"n/a"}
+                                }
+                                else {
+                                    td {+"${annotationsPerUser[user]?.minus(previousAnnotationsPerUser[user]!!)}"}
+                                }
+                            }
+                        }
+                    }
+                    table {
+                        tr {
+                            th {+"Subtype"}
+                            th {+"Annotations"}
+                            th {+"New annotations"}
+                        }
+                        for (subtype in sortedSubtypes.keys) {
+                            tr {
+                                td {+"$subtype" }
+                                td {+"${annotationsPerSubtype[subtype]}"}
+                                if (previousAnnotationsPerSubtype[subtype] == null) {
+                                    td {+"n/a"}
+                                }
+                                else {
+                                    td {+"${annotationsPerSubtype[subtype]?.minus(previousAnnotationsPerSubtype[subtype]!!)}"}
+                                }
+                            }
+                        }
+                    }
+                }
+            }
+            }
+        }
+        /**
+         * Search as far back as a week for a previous stats report
+         * The length of time can be changed
+         */
+        fun locatePreviousStats(currentDate: Calendar, statsDir: File): String {
+            val calendar = currentDate
```

Comment:
nit: This works, but a more elegant solution would be to take all the matching files in the directory, extract dates from them, sort them, and take the latest.